public/lib/aws_tools/s3_handler.py

```diff
@@ -0,0 +1,239 @@
+    #         else:
+    #             raise
+    #     else:
+    #         exists = True
+
+    #     return exists
+
+
+    # def key_modified_time(self, key, bucket_name=None):
+    #     """
+    #     get last modified time for key
+    #     :param key:
+    #     :param bucket_name:
+    #     :return:
+    #     """
+    #     if not bucket_name:
+    #         bucket = self.bucket
+    #     else:
+    #         bucket = self.resource.Bucket(bucket_name)
+
+    #     try:
+    #         s3_object = bucket.Object(key=key)
+    #     except botocore.exceptions.ClientError as e:
+    #         if e.response['Error']['Code'] == "404":
+    #             return None
+    #         else:
+    #             raise
+
+    #     return s3_object.last_modified
+
+
+    # def copy(self, from_key, from_bucket=None, to_key=None, catch_exception=True):
+    #     if not to_key:
+    #         to_key = from_key
+    #     if not from_bucket:
+    #         from_bucket = self.bucket_name
+
+    #     if catch_exception:
+    #         try:
+    #             return self.resource.Object(bucket_name=self.bucket_name, key=to_key).copy_from(
+    #                 CopySource='{0}/{1}'.format(from_bucket, from_key))
+    #         except:
+    #             return False
+    #     else:
+    #         return self.resource.Object(bucket_name=self.bucket_name, key=to_key).copy_from(
+    #             CopySource='{0}/{1}'.format(from_bucket, from_key))
+
+
+    # def replace(self, key, catch_exception=True):
+    #     if catch_exception:
+    #         try:
+    #             return self.resource.Object(bucket_name=self.bucket_name, key=key).copy_from(
+    #                 CopySource='{0}/{1}'.format(self.bucket_name, key), MetadataDirective='REPLACE')
+    #         except:
+    #             return False
+    #     else:
+    #         return self.resource.Object(bucket_name=self.bucket_name, key=key).copy_from(
+    #             CopySource='{0}/{1}'.format(self.bucket_name, key), MetadataDirective='REPLACE')
+
+
+    def upload_file(self, path, key, cache_time=600, content_type=None):
+        """
+        Upload file to S3 storage. Similar to the s3.upload_file, however, that
+        does not work nicely with moto, whereas this function does.
+        :param string path: file to upload
+        :param string key: name of the object in the bucket
+        """
+        from lib.general_tools.file_utils import get_mime_type
+        #GlobalSettings.logger.debug(f"s3_handler.upload_file({path}, {key}, {cache_time}, {content_type})")
+        assert 'http' not in key.lower()
+
+        with open(path, 'rb') as f:
+            binary = f.read()
+        if content_type is None:
+            content_type = get_mime_type(path)
+        self.bucket.put_object(
+            Key=key,
+            Body=binary,
+            ContentType=content_type,
+            CacheControl='max-age={0}'.format(cache_time)
+        )
+
+
+    # def get_object(self, key):
```

Comment:
I'm guessing this was copied from another project. I would just remove the methods that are commented out, that way the next person isn't trying to figure out if they are part of a feature that hasn't been implemented yet.